TCD1304.py

```diff
@@ -76,14 +73,17 @@ def master_clock (self):        # puts PWM with frequency of 2MHz on SQ1; master
     def sh_clock (self):           #  on SQ2 sets pulse for the integrationtime. Running the Sensor in shutter mode try with tint (min) = 10µs
         self.pwmgen.generate(["SQ2"], 1/integration_time, [0.5] ) 
 
+    def analog_signal_read (self, block:False):
+        self.scope.select_range('CH1', max_voltage_output)   # voltagerange should be fitted to the sensors output for better resolution. sensor otput is between 2V-3V due to datasheet
+        self.scope.configure_trigger(channel = 'CH1', voltage = min_voltage_output ) # starts recording , when voltage is over a certain level
+        self.scope.capture(channels=1, samples=integration_elements*samples_per_element, timegap=integration_time/samples_per_element, block=False)
+        self.icg_clock(self)
+        x, y= self.scope.fetch_data()
+        diff = abs(x,y[1, 0] - min_voltage_output )
+        analog_measurement = np.ndarray ()
+        return analog_measurement  # this should put the Measurements and timestamps in an array

```

Comment:
Here you are creating an empty numpy array and returning it. You probably want to put something in it first.

Reply:
I wanted to put the fetched data in there so we can work with the data. What does 'fetch_data' put it's data in?

Reply:
It returns a list of numpy arrays. The length of the list is equal to the number of channels in use, so one in your case.

I also just remembered that `fetch_data` does not return the time vector `x`, only the associated samples `y`. `x` is returned from `capture`, even in non-blocking mode.

To get both the time vector and collected samples as numpy arrays, you can do:
```python
x, = self.scope.capture(...)  # Note the comma after the x. It makes it so the first element of the returned list is put into x, instead of putting the list itself into x.
...
y, = self.scope.fetch_data()  # Again, note the comma.
```

Reply:
Do we need the time vector? I think it's not relevant for us, cause we have to sort the measurements to the different wavelength later on so we only have to pick the samples.